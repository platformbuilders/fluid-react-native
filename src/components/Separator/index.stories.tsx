import type { Meta } from '@storybook/react';
import Separator from '.';

const meta: Meta<typeof Separator> = {
  title: 'Separator',
  component: Separator,
  args: {
    marginBottom: 10,
    marginTop: 10,
    text: 'Texto de exemplo',
  },
  parameters: {
    notes:
      'Componente para renderização de um texto com uma linha horizontal embaixo.',
  },
};

export default meta;

export const Default = {};
