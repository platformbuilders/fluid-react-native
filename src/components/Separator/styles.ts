import { StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import { getTheme } from '@platformbuilders/theme-toolkit';
import Typography from '../Typography';

const smallerSpacing = getTheme('spacing.xs');
const disabled = getTheme('text.light');

type Props = {
  marginTop?: number;
  marginBottom?: number;
};

export const OptionsWrapper = styled.View`
  margin-top: ${({ marginTop }: Props) => marginTop || 0}px;
  margin-bottom: ${({ marginBottom }: Props) => marginBottom || 0}px;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

export const Line = styled.View`
  width: 33%;
  height: ${StyleSheet.hairlineWidth}px;
  background-color: ${disabled}90;
`;

export const DividerText = styled(Typography)`
  padding: ${smallerSpacing}px;
  color: ${disabled}90;
  font-weight: bold;
`;
