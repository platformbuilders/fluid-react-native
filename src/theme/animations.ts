import { Animations } from '@platformbuilders/theme-toolkit';
import {
  LoadDotsAnimation,
  LoadDotsContrastAnimation,
  LoadingCircleAnimation,
  LoadingCircleContrastAnimation,
  LoadingProgressAnimation,
} from '../assets/animations';

export default {
  circularLoading: LoadingCircleAnimation,
  contrastCircularLoading: LoadingCircleContrastAnimation,
  buttonLoading: LoadDotsAnimation,
  contrastButtonLoading: LoadDotsContrastAnimation,
  linearLoading: LoadingProgressAnimation,
  contrastLinearLoading: LoadingProgressAnimation,
} as Animations;
