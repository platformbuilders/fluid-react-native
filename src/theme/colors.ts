import { darken, lighten } from 'polished';
import { Colors } from '../types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';
const disabled = '#eeeeee';

const colors: Colors = {
  text: '#212121',
  info: '#4096D1',
  success: '#357a38',
  warning: '#F5B800',
  failure: '#cc0000',
  disabled: {
    light: lighten(0.05, disabled),
    main: disabled,
    dark: darken(0.3, disabled),
    contrast: '#cccccc',
  },
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
