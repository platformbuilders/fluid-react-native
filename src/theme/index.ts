export { default as colors } from './colors';
export { default as spacing } from './spacing';
export { default as radius } from './radius';
export { default as animations } from './animations';
export { default as typography } from './typography';
