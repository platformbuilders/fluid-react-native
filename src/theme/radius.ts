import { moderateScale } from 'react-native-size-matters';
import { Radius } from '../types';

export default {
  smallRadius: `${moderateScale(3)}px`,
  mediumRadius: `${moderateScale(8)}px`,
  largeRadius: `${moderateScale(15)}px`,
  modalRadius: `${moderateScale(20)}px`,
  buttonRadius: `${moderateScale(7)}px`,
} as Radius;
