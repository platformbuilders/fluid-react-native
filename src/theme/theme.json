{
  "spacing": {
    "min": {
      "value": 1,
      "type": "spacing"
    },
    "xxs": {
      "value": 2,
      "type": "spacing"
    },
    "xs": {
      "value": 4,
      "type": "spacing"
    },
    "sm": {
      "value": 8,
      "type": "spacing"
    },
    "md": {
      "value": 16,
      "type": "spacing"
    },
    "lg": {
      "value": 24,
      "type": "spacing"
    },
    "xl": {
      "value": 32,
      "type": "spacing"
    },
    "xxl": {
      "value": 48,
      "type": "spacing"
    },
    "max": {
      "value": 64,
      "type": "spacing"
    }
  },
  "sizing": {
    "min": {
      "value": 1,
      "type": "sizing"
    },
    "xxs": {
      "value": 2,
      "type": "sizing"
    },
    "xs": {
      "value": 4,
      "type": "sizing"
    },
    "sm": {
      "value": 8,
      "type": "sizing"
    },
    "md": {
      "value": 16,
      "type": "sizing"
    },
    "lg": {
      "value": 24,
      "type": "sizing"
    },
    "xl": {
      "value": 32,
      "type": "sizing"
    },
    "xxl": {
      "value": 48,
      "type": "sizing"
    },
    "max": {
      "value": 64,
      "type": "sizing"
    }
  },
  "borderRadius": {
    "min": {
      "value": 0,
      "type": "borderRadius"
    },
    "xxs": {
      "value": 1,
      "type": "borderRadius"
    },
    "xs": {
      "value": 2,
      "type": "borderRadius"
    },
    "sm": {
      "value": 4,
      "type": "borderRadius"
    },
    "md": {
      "value": 6,
      "type": "borderRadius"
    },
    "lg": {
      "value": 8,
      "type": "borderRadius"
    },
    "xl": {
      "value": 16,
      "type": "borderRadius"
    },
    "xxl": {
      "value": 32,
      "type": "borderRadius"
    },
    "max": {
      "value": 999,
      "type": "borderRadius"
    }
  },
  "borderWidth": {
    "min": {
      "value": 0.5,
      "type": "borderWidth"
    },
    "xxs": {
      "value": 1,
      "type": "borderWidth"
    },
    "xs": {
      "value": 1.5,
      "type": "borderWidth"
    },
    "sm": {
      "value": 2,
      "type": "borderWidth"
    },
    "md": {
      "value": 3,
      "type": "borderWidth"
    },
    "lg": {
      "value": 4,
      "type": "borderWidth"
    },
    "xl": {
      "value": 6,
      "type": "borderWidth"
    },
    "xxl": {
      "value": 8,
      "type": "borderWidth"
    },
    "max": {
      "value": 12,
      "type": "borderWidth"
    }
  },
  "boxShadow": {
    "z0": {
      "value": {
        "x": 0,
        "y": 0,
        "spread": 0,
        "color": "",
        "blur": 0
      },
      "type": "boxShadow"
    },
    "z1": {
      "value": {
        "x": 0,
        "y": 1,
        "spread": 0,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 3
      },
      "type": "boxShadow"
    },
    "z2": {
      "value": {
        "x": 0,
        "y": 4,
        "spread": -1,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 6
      },
      "type": "boxShadow"
    },
    "z3": {
      "value": {
        "x": 0,
        "y": 10,
        "spread": -3,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 15
      },
      "type": "boxShadow"
    },
    "z4": {
      "value": {
        "x": 0,
        "y": 20,
        "spread": -5,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 25
      },
      "type": "boxShadow"
    },
    "z5": {
      "value": {
        "x": 0,
        "y": 25,
        "spread": -12,
        "color": "rgba(0, 0, 0, 0.2)",
        "blur": 50
      },
      "type": "boxShadow"
    }
  },
  "brand": {
    "primary": {
      "light": {
        "value": "#7684F9",
        "type": "color"
      },
      "main": {
        "value": "#4255EF",
        "type": "color"
      },
      "dark": {
        "value": "#1B288E",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "secondary": {
      "light": {
        "value": "#8BD6F8",
        "type": "color"
      },
      "main": {
        "value": "#42ADEF",
        "type": "color"
      },
      "dark": {
        "value": "#337BB7",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "tertiary": {
      "light": {
        "value": "#BA96F5",
        "type": "color"
      },
      "main": {
        "value": "#8442EF",
        "type": "color"
      },
      "dark": {
        "value": "#4A00DC",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "accent": {
      "light": {
        "value": "#FBD268",
        "type": "color"
      },
      "main": {
        "value": "#F49D19",
        "type": "color"
      },
      "dark": {
        "value": "#A1530A",
        "type": "color"
      },
      "contrast": {
        "value": "#351B02",
        "type": "color"
      }
    }
  },
  "success": {
    "light": {
      "value": "#F4FCD1",
      "type": "color"
    },
    "main": {
      "value": "#96C920",
      "type": "color"
    },
    "dark": {
      "value": "#396006",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "danger": {
    "light": {
      "value": "#FFE8D8",
      "type": "color",
      "description": "the 100 shade of a color"
    },
    "main": {
      "value": "#FF4F3F",
      "type": "color"
    },
    "dark": {
      "value": "#7A0C27",
      "type": "color",
      "description": "the 900s shade of a color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "warning": {
    "light": {
      "value": "#FFF8CE",
      "type": "color"
    },
    "main": {
      "value": "#FFCA0C",
      "type": "color"
    },
    "dark": {
      "value": "#7A5502",
      "type": "color"
    },
    "contrast": {
      "value": "#261B00",
      "type": "color"
    }
  },
  "info": {
    "light": {
      "value": "#C2ECEC",
      "type": "color"
    },
    "main": {
      "value": "#42CCD4",
      "type": "color"
    },
    "dark": {
      "value": "#186F82",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "text": {
    "main": {
      "value": "#444444",
      "type": "color"
    },
    "dark": {
      "value": "#000",
      "type": "color"
    },
    "light": {
      "value": "#999999",
      "type": "color"
    },
    "contrast": {
      "value": "#fff",
      "type": "color"
    }
  },
  "background": {
    "z0": {
      "value": "#ffffff",
      "type": "color"
    },
    "z1": {
      "value": "#fafafa",
      "type": "color"
    },
    "z2": {
      "value": "#f0f0f0",
      "type": "color"
    },
    "z3": {
      "value": "#dddddd",
      "type": "color"
    },
    "z4": {
      "value": "#d1d1d1",
      "type": "color"
    }
  },
  "opacity": {
    "disabled": {
      "value": "30%",
      "type": "opacity"
    },
    "empty": {
      "value": "50%",
      "type": "opacity"
    },
    "overlay": {
      "value": "80%",
      "type": "opacity"
    }
  },
  "fontFamilies": {
    "heading": {
      "value": "Roboto",
      "type": "fontFamilies"
    },
    "body": {
      "value": "Roboto",
      "type": "fontFamilies"
    }
  },
  "lineHeights": {
    "heading": {
      "value": "110%",
      "type": "lineHeights"
    },
    "body": {
      "value": "140%",
      "type": "lineHeights"
    }
  },
  "letterSpacing": {
    "default": {
      "value": 0,
      "type": "letterSpacing"
    },
    "bigger": {
      "value": "105%",
      "type": "letterSpacing"
    }
  },
  "paragraphSpacing": {
    "h1": {
      "value": 32,
      "type": "paragraphSpacing"
    },
    "h2": {
      "value": 26,
      "type": "paragraphSpacing"
    }
  },
  "fontWeights": {
    "headingRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "headingBold": {
      "value": "Bold",
      "type": "fontWeights"
    },
    "bodyRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "bodyBold": {
      "value": "Bold",
      "type": "fontWeights"
    }
  },
  "fontSizes": {
    "min": {
      "value": 10,
      "type": "fontSizes"
    },
    "xxs": {
      "value": 11,
      "type": "fontSizes"
    },
    "xs": {
      "value": 13,
      "type": "fontSizes"
    },
    "sm": {
      "value": 14,
      "type": "fontSizes"
    },
    "md": {
      "value": 15,
      "type": "fontSizes"
    },
    "lg": {
      "value": 17,
      "type": "fontSizes"
    },
    "xl": {
      "value": 20,
      "type": "fontSizes"
    },
    "xxl": {
      "value": 24,
      "type": "fontSizes"
    },
    "max": {
      "value": 32,
      "type": "fontSizes"
    }
  },
  "themeRadius": {
    "button": {
      "value": 3,
      "type": "borderRadius"
    },
    "card": {
      "value": 6,
      "type": "borderRadius"
    },
    "input": {
      "value": 3,
      "type": "borderRadius"
    }
  },
  "typography": {
    "H1": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "H2": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "Body": {
      "type": "typography",
      "value": {
        "fontFamily": "$fontFamilies.body",
        "fontWeight": "$fontWeights.bodyRegular",
        "lineHeight": "$lineHeights.heading",
        "fontSize": "$fontSizes.body",
        "paragraphSpacing": "$paragraphSpacing.h2"
      }
    }
  }
}
