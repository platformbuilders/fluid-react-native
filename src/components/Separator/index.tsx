import React from 'react';
import { DividerText, Line, OptionsWrapper } from './styles';

type Props = {
  marginTop?: number;
  marginBottom?: number;
  text: string;
};

const Separator: React.FC<Props> = ({ text, marginTop, marginBottom }) => (
  <OptionsWrapper marginTop={marginTop} marginBottom={marginBottom}>
    <Line />
    <DividerText>{text}</DividerText>
    <Line />
  </OptionsWrapper>
);

export default Separator;
