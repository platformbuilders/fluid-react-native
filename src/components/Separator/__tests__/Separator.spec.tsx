import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import { render } from '@testing-library/react-native';
import Separator from '..';
import theme from '../../../theme';

describe('<Separator />', () => {
  it('should render with required props', () => {
    let wrapper;

    act(() => {
      wrapper = renderer.create(
        <ThemeProvider theme={theme}>
          <Separator text="ou" />
        </ThemeProvider>,
      );
    });

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render with custom margins', () => {
    let wrapper;

    act(() => {
      wrapper = renderer.create(
        <ThemeProvider theme={theme}>
          <Separator text="ou" marginTop={20} marginBottom={10} />
        </ThemeProvider>,
      );
    });

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render with only marginTop', () => {
    let wrapper;

    act(() => {
      wrapper = renderer.create(
        <ThemeProvider theme={theme}>
          <Separator text="ou" marginTop={20} />
        </ThemeProvider>,
      );
    });

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render with only marginBottom', () => {
    let wrapper;

    act(() => {
      wrapper = renderer.create(
        <ThemeProvider theme={theme}>
          <Separator text="ou" marginBottom={10} />
        </ThemeProvider>,
      );
    });

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render with long text', () => {
    let wrapper;

    act(() => {
      wrapper = renderer.create(
        <ThemeProvider theme={theme}>
          <Separator text="texto longo para teste" />
        </ThemeProvider>,
      );
    });

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  // Testes adicionais para aumentar a cobertura

  it('should render the correct text content', () => {
    const testText = 'texto de teste';

    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Separator text={testText} />
      </ThemeProvider>,
    );

    const textElement = getByText(testText);
    expect(textElement).toBeTruthy();
    expect(textElement.props.children).toBe(testText);
  });

  it('should apply custom margins correctly', () => {
    const marginTop = 30;
    const marginBottom = 15;

    let wrapper;

    act(() => {
      wrapper = renderer.create(
        <ThemeProvider theme={theme}>
          <Separator
            text="custom margins"
            marginTop={marginTop}
            marginBottom={marginBottom}
          />
        </ThemeProvider>,
      );
    });

    const json = wrapper.toJSON();
    const containerStyles = json.props.style;

    // Verificar que os estilos de margem foram aplicados corretamente
    expect(containerStyles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          marginTop,
          marginBottom,
        }),
      ]),
    );
  });
});
