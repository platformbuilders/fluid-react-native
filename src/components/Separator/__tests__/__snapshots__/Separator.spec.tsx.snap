// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> should render with custom margins 1`] = `
<View
  marginBottom={10}
  marginTop={20}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
  <Text
    accessibilityLabel="ou"
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#99999990",
          "fontWeight": "bold",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    variant="md"
  >
    ou
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
</View>
`;

exports[`<Separator /> should render with long text 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
  <Text
    accessibilityLabel="texto longo para teste"
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#99999990",
          "fontWeight": "bold",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    variant="md"
  >
    texto longo para teste
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
</View>
`;

exports[`<Separator /> should render with only marginBottom 1`] = `
<View
  marginBottom={10}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
  <Text
    accessibilityLabel="ou"
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#99999990",
          "fontWeight": "bold",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    variant="md"
  >
    ou
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
</View>
`;

exports[`<Separator /> should render with only marginTop 1`] = `
<View
  marginTop={20}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
  <Text
    accessibilityLabel="ou"
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#99999990",
          "fontWeight": "bold",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    variant="md"
  >
    ou
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
</View>
`;

exports[`<Separator /> should render with required props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
  <Text
    accessibilityLabel="ou"
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#99999990",
          "fontWeight": "bold",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    variant="md"
  >
    ou
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 0.5,
          "width": "33%",
        },
      ]
    }
  />
</View>
`;
